indirect.h

```diff
@@ -66,15 +66,18 @@ class indirect {
   using pointer = typename allocator_traits::pointer;
   using const_pointer = typename allocator_traits::const_pointer;
 
+  template <typename U = T>
   constexpr indirect()
-    requires std::is_default_constructible_v<A>
+    requires(std::is_default_constructible_v<A> &&
+             std::is_default_constructible_v<U>)
   {
-    static_assert(std::is_default_constructible_v<T>);
     p_ = construct_from(alloc_);
   }
 
-  constexpr indirect(std::allocator_arg_t, const A& alloc) : alloc_(alloc) {
-    static_assert(std::is_default_constructible_v<T>);
+  template <typename U = T>
+  constexpr indirect(std::allocator_arg_t, const A& alloc)
+    requires std::is_default_constructible_v<U>
```

Comment:
Perhaps we should be checking that the allocator is copy-constructible here too?

Reply:
From https://en.cppreference.com/w/cpp/named_req/Allocator this is a requirement of any allocator. It just needs to satisfy the _Cpp17AllocatorRequirments_.

Nice catch though.

Reply:
Do we state allocator requirements anywhere?

Reply:
The front matter has a requirement that Alloc satisfy Cpp17Allocator concept, I believe.

Reply:
> The front matter has a requirement that Alloc satisfy Cpp17Allocator concept, I believe.

Hmm, no it seems it's not there. I have added #227 to move this requirement to the front-matter.